rust_src/src/fns.rs

```diff
@@ -375,4 +383,61 @@ pub fn load_average(use_floats: bool) -> Vec<LispNumber> {
         .collect()
 }
 
+/// Ask user a yes-or-no question.
+///
+/// Return t if answer is yes, and nil if the answer is no.  PROMPT is
+/// the string to display to ask the question.  It should end in a
+/// space; `yes-or-no-p' adds \"(yes or no) \" to it.
+///
+/// The user must confirm the answer with RET, and can edit it until
+/// it has been confirmed.
+///
+/// If dialog boxes are supported, a dialog box will be used if
+/// `last-nonmenu-event' is nil, and `use-dialog-box' is non-nil.
+#[lisp_fn]
+pub fn yes_or_no_p(prompt: LispStringRef) -> bool {
+    let use_popup = unsafe {
+        (globals.last_nonmenu_event.is_nil() || globals.last_nonmenu_event.is_cons())
+            && globals.use_dialog_box
+            && !globals.last_input_event.is_nil()
+    };
+
+    if use_popup {
+        unsafe { redisplay_preserve_echo_area(4) };
+        let pane = list![LispObject::cons("Yes", Qt), LispObject::cons("No", Qnil)];
```

Comment:
@nickdrozd added support for a 2-tuple of LispObjects to be converted to a Cons cell.

```suggestion
        let pane = list![("Yes", Qt).into(), ("No", Qnil).into()];
```

Reply:
It turns out this won't compile,

    error[E0283]: type annotations required: cannot resolve `_: std::convert::Into<lisp::LispObject>`

and I'm not finding a way to do this in one expression. It does work if I extract each into its own

    let a: LispObject = ("Yes", Qt).into();

but that seems less readable than before.